### PR TITLE
Add MetadataClient getCommittedOffsets

### PR DESCRIPTION
**Pull request by epalace**
Stop using deprecated KafkaConsumer#committed method, and start using its replacement. This allows to use a set of partitions in a single call, which has better performance than iterating on the old deprecated method.

Old PR: https://github.com/akka/alpakka-kafka/pull/1071

### CODE REVIEW

File: core/src/main/scala/akka/kafka/Metadata.scala [hunk 1]
```diff
@@ -185,5 +188,34 @@ object Metadata {
...
+
+    /**
+     * Java API
+     */
+    def getResponse: Optional[java.util.Map[TopicPartition, OffsetAndMetadata]] =
+      response
+        .map { m =>
+          Optional.of(m.asJava)
+        }
+        .getOrElse(Optional.empty())
```
Comment (seglo) [inline core/src/main/scala/akka/kafka/Metadata.scala]:
```suggestion
      Optional.ofNullable(response.toOption.map(_.asJava).orNull)
```

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by epalace** [comment]
@seglo could you have a last look to it?

**Comment by seglo** [comment]
> @seglo could you have a last look to it?

I'll take a look shortly.  In the meantime do you mind squashing your commits with your GitHub user id that signed the CLA?  Then the `lightbend-cla-validator` will stop complaining.

**Comment by ennru** [comment]
Thank you!

**Pull request merged**